### ISSUE TITLE
Stuck in Building model on training data

### ISSUE THREAD

**Issue by sangyx** [Waikato/wekaDeeplearning4j#76]
**Describe the bug**
When I tried to follow the tutorial ["Starting Simple - The Iris Dataset (Tabular)"](https://deeplearning.cms.waikato.ac.nz/examples/train_cnn/). It stucked in the Status "Building model on training data" after I clicked "start".

**Screenshots**
<img width="1021" alt="image" src="https://github.com/Waikato/wekaDeeplearning4j/assets/32364921/b34b451b-5d9e-480c-83dd-4b278652ed64">

log:
```
01:29:03: Weka Explorer
01:29:03: (c) 1999-2022 The University of Waikato, Hamilton, New Zealand
01:29:03: web: https://www.cs.waikato.ac.nz/~ml/weka/
01:29:03: Started on Monday, 6 May 2024
01:29:19: Base relation is now iris (150 instances)
01:31:55: Started weka.classifiers.functions.Dl4jMlpClassifier
01:31:55: Command: weka.classifiers.functions.Dl4jMlpClassifier -S 1 -cache-mode MEMORY -early-stopping "weka.dl4j.earlystopping.EarlyStopping -maxEpochsNoImprovement 0 -valPercentage 0.0" -normalization "Standardize training data" -iterator "weka.dl4j.iterators.instance.DefaultInstanceIterator -bs 1" -iteration-listener "weka.dl4j.listener.EpochListener -eval true -n 5" -layer "weka.dl4j.layers.OutputLayer -lossFn \"weka.dl4j.lossfunctions.LossMCXENT \" -nOut 2 -activation \"weka.dl4j.activations.ActivationSoftmax \" -name \"Output layer\"" -logConfig "weka.core.LogConfiguration -append true -dl4jLogLevel WARN -logFile /Users/ys26425/wekafiles/wekaDeeplearning4j.log -nd4jLogLevel INFO -wekaDl4jLogLevel INFO" -config "weka.dl4j.NeuralNetConfiguration -biasInit 0.0 -biasUpdater \"weka.dl4j.updater.Sgd -lr 0.001 -lrSchedule \\\"weka.dl4j.schedules.ConstantSchedule -scheduleType EPOCH\\\"\" -dist \"weka.dl4j.distribution.Disabled \" -dropout \"weka.dl4j.dropout.Disabled \" -gradientNormalization None -gradNormThreshold 1.0 -l1 NaN -l2 NaN -minimize -algorithm STOCHASTIC_GRADIENT_DESCENT -updater \"weka.dl4j.updater.Adam -beta1MeanDecay 0.9 -beta2VarDecay 0.999 -epsilon 1.0E-8 -lr 0.001 -lrSchedule \\\"weka.dl4j.schedules.ConstantSchedule -scheduleType EPOCH\\\"\" -weightInit XAVIER -weightNoise \"weka.dl4j.weightnoise.Disabled \"" -numEpochs 10 -numGPUs 1 -averagingFrequency 10 -prefetchSize 24 -queueSize 0 -zooModel "weka.dl4j.zoo.CustomNet -channelsLast false -pretrained NONE"
```

**Additional Information**

- Weka version: 3.9.6
- wekaDeeplearning4j package version: Install in Package Manager, 1.7.2
- Operating System: macOS Sonoma Version 14.4.1  - M2 Pro

**Additional context**
I also can't see the result of "Is CPU Available?"

<img width="1022" alt="image" src="https://github.com/Waikato/wekaDeeplearning4j/assets/32364921/ccbca9c6-bb1d-4356-a275-eeca20248db1">

**Comment by eibe** [comment]
Unfortunately, ARM chips are not supported by the old release of deeplearning4j that is used by WEKA. Updating the wekaDeepLearning4j package to use a more recent release of deeplearning4j is on the to-do list.

If you are happy to write Python code, another option for deep learning is the weka-pyscript package:

  https://github.com/christopher-beckham/weka-pyscript

Cheers,
Eibe

From: Yunxin Sang ***@***.***>
Date: Monday, 6 May 2024 at 6:40 PM
To: Waikato/wekaDeeplearning4j ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Waikato/wekaDeeplearning4j] Stuck on Building model on training data (Issue #76)

Describe the bug
When I tried to follow the tutorial "Starting Simple - The Iris Dataset (Tabular)"<https://deeplearning.cms.waikato.ac.nz/examples/train_cnn/>. When I click the "Start", it stucks in the Status "Building model on training data"

Screenshots
If applicable, add screenshots to help explain your problem.
image.png (view on web)<https://github.com/Waikato/wekaDeeplearning4j/assets/32364921/b34b451b-5d9e-480c-83dd-4b278652ed64>

log:

01:29:03: Weka Explorer

01:29:03: (c) 1999-2022 The University of Waikato, Hamilton, New Zealand

01:29:03: web: https://www.cs.waikato.ac.nz/~ml/weka/

01:29:03: Started on Monday, 6 May 2024

01:29:19: Base relation is now iris (150 instances)

01:31:55: Started weka.classifiers.functions.Dl4jMlpClassifier

01:31:55: Command: weka.classifiers.functions.Dl4jMlpClassifier -S 1 -cache-mode MEMORY -early-stopping "weka.dl4j.earlystopping.EarlyStopping -maxEpochsNoImprovement 0 -valPercentage 0.0" -normalization "Standardize training data" -iterator "weka.dl4j.iterators.instance.DefaultInstanceIterator -bs 1" -iteration-listener "weka.dl4j.listener.EpochListener -eval true -n 5" -layer "weka.dl4j.layers.OutputLayer -lossFn \"weka.dl4j.lossfunctions.LossMCXENT \" -nOut 2 -activation \"weka.dl4j.activations.ActivationSoftmax \" -name \"Output layer\"" -logConfig "weka.core.LogConfiguration -append true -dl4jLogLevel WARN -logFile /Users/ys26425/wekafiles/wekaDeeplearning4j.log -nd4jLogLevel INFO -wekaDl4jLogLevel INFO" -config "weka.dl4j.NeuralNetConfiguration -biasInit 0.0 -biasUpdater \"weka.dl4j.updater.Sgd -lr 0.001 -lrSchedule \\\"weka.dl4j.schedules.ConstantSchedule -scheduleType EPOCH\\\"\" -dist \"weka.dl4j.distribution.Disabled \" -dropout \"weka.dl4j.dropout.Disabled \" -gradientNormalization None -gradNormThreshold 1.0 -l1 NaN -l2 NaN -minimize -algorithm STOCHASTIC_GRADIENT_DESCENT -updater \"weka.dl4j.updater.Adam -beta1MeanDecay 0.9 -beta2VarDecay 0.999 -epsilon 1.0E-8 -lr 0.001 -lrSchedule \\\"weka.dl4j.schedules.ConstantSchedule -scheduleType EPOCH\\\"\" -weightInit XAVIER -weightNoise \"weka.dl4j.weightnoise.Disabled \"" -numEpochs 10 -numGPUs 1 -averagingFrequency 10 -prefetchSize 24 -queueSize 0 -zooModel "weka.dl4j.zoo.CustomNet -channelsLast false -pretrained NONE"

Additional Information

  *   Weka version: 3.9.6
  *   wekaDeeplearning4j package version: Install in Package Manager, 1.7.2
  *   Operating System: macOS Sonoma Version 14.4.1 - M2 Pro

Additional context
Add any other context about the problem here.
I also can't see the result of "Is CPU Available?"
image.png (view on web)<https://github.com/Waikato/wekaDeeplearning4j/assets/32364921/ccbca9c6-bb1d-4356-a275-eeca20248db1>

**Comment by sangyx** [comment]
@eibe Thanks for your reply. Do you have a timeline for updating wekaDeeplearning4j? We're using Weka in an MBA class and trying to add more advanced content like CNN into our class.

**Comment by eibe** [comment]
The plan is to have this update done by the end of June.

Cheers,
Eibe

From: Yunxin Sang ***@***.***>
Date: Thursday, 9 May 2024 at 2:29 PM
To: Waikato/wekaDeeplearning4j ***@***.***>
Cc: Eibe Frank ***@***.***>, Mention ***@***.***>
Subject: Re: [Waikato/wekaDeeplearning4j] Stuck in Building model on training data (Issue #76)

@eibe<https://github.com/eibe> Thanks for your reply. Do you have a timeline for updating wekaDeeplearning4j? We're using Weka in an MBA class and trying to add more advanced content like CNN into our class.
